src/packager.ts

```diff
@@ -1,3 +1,153 @@
+import fs from "fs/promises";
+import os from "os";
+import path from "path";
+import stream from "stream";
+import {
+  GetObjectCommand,
+  HeadObjectCommand,
+  NotFound,
+  S3Client,
+  paginateListObjectsV2,
+} from "@aws-sdk/client-s3";
+import { Upload } from "@aws-sdk/lib-storage";
 import { ScheduledHandler } from "aws-lambda";
+import { create as struct } from "superstruct";
+import { create as createTar } from "tar";
+import { localFromISO } from "./datetime";
+import { Environment } from "./packager/structs";
 
-export const handler: ScheduledHandler = async () => undefined;
+class BadS3ResponseError extends Error {}
+class OutputKeyExistsError extends Error {}
+
+export const handler: ScheduledHandler = async ({ time }) => {
+  // Copy the env to avoid `optional` properties becoming the string "undefined"
+  // instead of real `undefined`; this may have to do with the automatic string
+  // conversion behavior of `process.env`
+  const env = struct({ ...process.env }, Environment);
+  const client = makeS3Client(env);
+  const {
+    S3_BUCKET: bucket,
+    S3_PREFIX_OUTPUT: outputPrefix,
+    S3_PREFIX_SOURCE: sourcePrefix,
+  } = env;
+
+  const serviceDay = localFromISO(time).minus({ days: 1 }).toISODate();
+  const shortServiceDay = serviceDay.replace(/-/g, "");
+  const outputKey = path.posix.join(outputPrefix, `${shortServiceDay}.tar.gz`);
+
+  if (await objectExists(client, bucket, outputKey))
+    throw new OutputKeyExistsError(outputKey);
```

Comment:
Maybe `{ }` here? I'm surprised this isn't a lint actually, but I thought single line `if` statements without braces were generally frowned upon in most languages. There was some Apple security issue years back where an additional statement was added without adding braces, causing it to unconditionally execute, but I forget the details of that.

Reply:
I think this isn't as much of a concern in this context because Prettier makes it impossible to "look like" you have a brace block when you don't. i.o.w. if you write the following:

```js
if (await objectExists(client, bucket, outputKey))
  const message = `Key exists: ${outputKey}`;
  throw new OutputKeyExistsError(message);
```

It will be reformatted to:

```js
if (await objectExists(client, bucket, outputKey))
  const message = `Key exists: ${outputKey}`;
throw new OutputKeyExistsError(message);
```

And this formatting makes it clear that the throw is unconditional.